pkg/controller/node/node_controller.go

```diff
@@ -97,7 +107,9 @@ func New(
 		UpdateFunc: ctrl.updateNode,
 		DeleteFunc: ctrl.deleteNode,
 	})
-
+	schedulerInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: ctrl.updateMasterNodes,
```

Comment:
I've a doubt and currently thinking Colin meant to wire all this into the current `sync` handler to couple this with the other upgrade decisions (but this might as well work out correctly)

Reply:
Yeah...indeed that was my original thought; if they're in separate handlers then it's possible that at least transiently we are violating the max schedulable.  But OTOH Kube is all about reconciliation/eventual consistency and the window for the labels being out of sync would likely be very small.

Reply:
I thought about this but in that case, we need to have scheduler lister list CRs continuously. My past experience with this CR listing continuously was that it causes a performance impact(a hot read loop). Also, out of sync can happen in even in that case where the update happened and sync loop just completed, we'd have to wait for next sync loop for changes to come into effect but the eventual consistency model as @cgwalters pointed out would still hold good :).